src/main.rs

```diff
@@ -109,6 +172,33 @@ fn main() -> Result<()> {
     let mut buffer = Vec::new();
     input.read_to_end(&mut buffer)?;
 
+    let schema_string_read_result = opts.read_schema_to_string();
+    let query_string_read_result = opts.read_query_to_string();
+    let input_json: serde_json::Value = serde_json::from_slice(&buffer)?;
```

Comment:
Should we map this error to provide a nicer message indicating that we could not parse the input as JSON?